tests/test_views.py

```diff
@@ -608,3 +613,115 @@ async def test_invoice_ws(async_client, token: str):
+    assert {
+        "price": 0.5,
+        "store_id": 2,
+        "discount": discount_id,
+        "products": [product_id],
+    }.items() < invoice_resp.json().items()
+    assert (
+        client.delete(f"/invoices/{invoice_resp.json()['id']}", headers={"Authorization": f"Bearer {token}"}).status_code
+        == 200
+    )
+
+
+@pytest.mark.parametrize("order_id,expect_status_code", [(-1, 404), (10, 200)])
+def test_get_invoice_by_order_id(client: TestClient, token: str, order_id: int, expect_status_code):
+    resp = None
+    if expect_status_code == 200:
+        resp = client.post(
+            "/invoices",
+            json={"store_id": 2, "price": 0.98, "order_id": str(order_id)},
+            headers={"Authorization": f"Bearer {token}"},
+        )
+        assert resp.status_code == 200
+    assert (
+        client.get(f"/invoices/order_id/{order_id}", headers={"Authorization": f"Bearer {token}"}).status_code
+        == expect_status_code
+    )
+    if resp:
+        assert client.delete(f"/invoices/{resp.json()['id']}", headers={"Authorization": f"Bearer {token}"}).status_code == 200
+
+
+def test_get_max_product_price(client: TestClient, token: str):
+    # create product
+    price = 999999.0
+    new_product = {"name": "apple", "price": price, "quantity": 1.0, "store_id": 2}
+    create_product_resp = client.post(
+        "/products", data={"data": json_module.dumps(new_product)}, headers={"Authorization": f"Bearer {token}"}
+    )
+    assert create_product_resp.status_code == 200
+    maxprice_resp = client.get(f"/products/maxprice?store={2}", headers={"Authorization": f"Bearer {token}"})
+    assert maxprice_resp.status_code == 200
+    assert maxprice_resp.json() == price
+
+
+def test_create_product_with_image(client: TestClient, token: str):
+    image = (
+        b"\x89PNG\r\n\x1a\n\x00\x00\x00\rIHDR\x00\x00\x00\x01\x00\x00\x00\x01\x08\x02\x00\x00\x00\x90wS"
+        b"\xde\x00\x00\x00\x0cIDATx\x9cc```\x00\x00\x00\x04\x00\x01\xf6\x178U\x00\x00\x00\x00IEND\xaeB`\x82"
+    )
+    new_product = {"name": "sunflower", "price": 0.1, "quantity": 1.0, "store_id": 2}
+    # post
+    create_product_resp = client.post(
+        "/products",
+        data={"data": json_module.dumps(new_product)},
+        files={"image": image},
+        headers={"Authorization": f"Bearer {token}"},
+    )
+    assert create_product_resp.status_code == 200
+    product_dict = create_product_resp.json()
+    # patch
+    patch_product_resp = client.patch(
+        f"/products/{product_dict['id']}",
+        data={
+            "data": json_module.dumps(
+                {"price": 0.15, "quantity": 2.0, "user_id": product_dict["user_id"], "name": "sunflower"}
+            )
+        },
+        headers={"Authorization": f"Bearer {token}"},
+    )
+    assert patch_product_resp.status_code == 200
+    # put
+    put_product_data = {
+        "price": 0.01,
+        "quantity": 1,
+        "name": "banana",
+        "download_url": "",
```

Comment:
You can use a smaller put product data, as some fields are populated with defaults, see
https://github.com/MrNaif2018/bitcart/blob/master/tests/fixtures/products.json#L247